### PR TITLE
chore: move nfc to receive screen

### PR DESCRIPTION
**Pull request by sandipndev**
Moves NFC stuff to Receive Screen.

image

### CODE REVIEW

File: app/components/modal-nfc/modal-nfc.tsx [hunk 1]
```diff
@@ -122,38 +146,42 @@ export const ModalNfc: React.FC<{
...
+
+          receiveViaNFC(destination, toBtcMoneyAmount(settlementAmount.amount))
         }
       }
 
       dismiss()
     }
 
     init()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (nicolasburtey) [inline app/components/modal-nfc/modal-nfc.tsx:174]: can you add a comment on why this is necessary? this has been a source of bug in the past

File: app/components/modal-nfc/modal-nfc.tsx [hunk 2]
```diff
@@ -122,38 +146,42 @@ export const ModalNfc: React.FC<{
...
+            amount = convertMoneyAmount(
+              toUsdMoneyAmount(settlementAmount.amount),
+              WalletCurrency.Btc,
+            ).amount
+          }
+
+          destination.validDestination.minWithdrawable = amount * 1000 // coz msats
+          destination.validDestination.maxWithdrawable = amount * 1000 // coz msats
+
+          console.log(destination)
```
Comment (nicolasburtey) [inline app/components/modal-nfc/modal-nfc.tsx:164]:
```suggestion
```

File: app/screens/receive-bitcoin-screen/use-receive-bitcoin.ts [hunk 3]
```diff
@@ -455,6 +457,56 @@ export const useReceiveBitcoin = () => {
...
+        input: {
+          walletId: prcd.receivingWalletDescriptor.id,
+          amount: settlementAmount.amount,
+          memo: prcd.memo || defaultDescription,
+        },
+      },
+    })
+
+    if (!data) {
+      throw new Error("No data returned from lnInvoiceCreate")
```
Comment (nicolasburtey) [inline app/screens/receive-bitcoin-screen/use-receive-bitcoin.ts:476]: is this error handled properly upstream? or could this crash the mobile app?

Comment (sandipndev) [reply app/screens/receive-bitcoin-screen/use-receive-bitcoin.ts]: I made this to display alert instead

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**